packages/core/src/agent/message-list/index.ts

```
@@ -2852,6 +2604,315 @@ export class MessageList {
+        text: coreMessage.content,
+        // Preserve providerOptions from ModelMessage level (e.g., system messages with cacheControl)
+        ...('providerOptions' in coreMessage && coreMessage.providerOptions
+          ? { providerMetadata: coreMessage.providerOptions }
+          : {}),
+      });
+    } else if (Array.isArray(coreMessage.content)) {
+      for (const part of coreMessage.content) {
+        switch (part.type) {
+          case 'text':
+            // Add step-start only after tool results for assistant messages
+            const prevPart = parts.at(-1);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)